tools/tpm2_changeauth.c

```diff
@@ -112,8 +112,13 @@ bool tpm2_tool_onstart(tpm2_options **opts) {
 
 static inline bool object_needs_parent(tpm2_loaded_object *obj) {
 
-    return (obj->handle & TPM2_HR_TRANSIENT)
-            || (obj->handle & TPM2_HR_PERSISTENT);
+    switch (obj->handle & TPM2_HR_RANGE_MASK) {
```

Comment:
Don't do this with a switch....your adding a bunch of jumps for something that could be optimized to just bit wise logic and likely all in the register file. I don't think the compiler will be able to optimize a switch as well. Just add the mask:

```C
TPM2_HC h = obj->handle & TPM2_HR_RANGE_MASK;
return (h & TPM2_HR_TRANSIENT)
            || (h & TPM2_HR_PERSISTENT);
```